src/events.jl

```diff
@@ -1,85 +1,694 @@
 
-Returns an [`EventList`](@ref) containing the extracted data.
+# Examples
+```julia
+# Basic usage
+ev = readevents("events.fits")
 
-## Notes
+# With custom options
+ev = readevents("events.fits", hdu=3, sort=true, T=Float32)
 
-The function extracts `TIME` and `ENERGY` columns from any TableHDU in the FITS
-file. All headers from each HDU are collected into the metadata field.
+# Reading extra columns
+ev = readevents("events.fits", extra_columns=["RAWX", "RAWY", "DETX", "DETY"])
+
+# Accessing the data
+println("Number of events: \$(length(ev))")
+println("Time range: \$(extrema(times(ev)))")
+if has_energies(ev)
+    println("Energy range: \$(extrema(energies(ev)))")
+    println("Energy column: \$(ev.meta.energy_units)")
+end
+```
+
+# Type Stability
+This function is designed to be type-stable with proper type annotations
+on return values from FITS reading operations. The return type is fully
+specified to enable compiler optimizations.
+
+# Error Handling
+- Throws `AssertionError` if time and energy vectors have different sizes
+- Throws `AssertionError` if times are not sorted and `sort=false`
+- FITS reading errors are propagated from the FITSIO.jl library
+
+# Implementation Notes
+- Uses type-stable FITS reading with explicit type conversions
+- Handles missing energy data gracefully
+- Supports efficient multi-column sorting when `sort=true`
+- Creates metadata with all relevant file information
+- Validates data consistency before returning
+- Added case_sensitive=false parameter: This tells FITSIO.jl to use the old behavior for backward compatibility
 """
-function readevents(path; T = Float64)
-    headers = Dict{String,Any}[]
-    times = T[]
-    energies = T[]
+function readevents(
+    path::AbstractString;
+    hdu::Int = 2,
+    T::Type = Float64,
+    sort::Bool = false,
+    extra_columns::Vector{String} = String[],
+    energy_alternatives::Vector{String} = ["ENERGY", "PI", "PHA"],
+    kwargs...,
+)::EventList{Vector{T},FITSMetadata{FITSIO.FITSHeader}}
+
+    # Read data from FITS file with type-stable operations
+    time::Vector{T},
+    energy::Union{Nothing,Vector{T}},
+    energy_col::Union{Nothing,String},
+    header::FITSIO.FITSHeader,
+    extra_data::Dict{String,Vector} = FITS(path, "r") do f
+
+        selected_hdu = f[hdu]
+
+        # Read header (type-stable)
+        header = read_header(selected_hdu)
+
+        # Get actual column names to find the correct TIME column
+        all_cols = FITSIO.colnames(selected_hdu)
+        time_col = findfirst(col -> uppercase(col) == "TIME", all_cols)
+        
+        if isnothing(time_col)
+            error("TIME column not found in HDU $hdu")
+        end
+        
+        actual_time_col = all_cols[time_col]
+        
+        # Read time column with case-insensitive option
+        time = convert(Vector{T}, read(selected_hdu, actual_time_col, case_sensitive=false))
```

Comment:
```suggestion
        time = convert(Vector{T}, read(selected_hdu, "TIME", case_sensitive=false))
```

What's wrong with this? It will raise an error if no time column is found already, and is less code to reason about.